snappy-tools/sbin/collect-debug-artifacts.sh

```diff
@@ -0,0 +1,230 @@
+  fi
+
+  if [ ! -f "${CONF_FILE}" ]; then
+    echo "Config file ${CONF_FILE} does not exists"
+    exit 1
+  fi
+
+  source $CONF_FILE
+
+  if [ ! -f "${MEMBERS_INFO_FILE}" ]; then
+    echo "MEMBERS_INFO_FILE file ${MEMBERS_INFO_FILE} does not exists"
+    exit 1
+  fi
+
+  if [ -z "${NO_OF_STACK_DUMPS}" ]; then
+    NO_OF_STACK_DUMPS=5
+  fi
+
+  if [ -z "${INTERVAL_BETWEEN_DUMPS}" ]; then
+    INTERVAL_BETWEEN_DUMPS=5
+  fi
+
+  if [ -z "${GET_EVERYTHING}" ]; then
+    GET_EVERYTHING=0
+  fi
+
+  if [ "${VERBOSE}" = "1" ]; then
+    echo CONF=$CONF_FILE
+    echo MEMINFO=$MEMBERS_INFO_FILE
+    echo NUM_STACK_DUMPS=$NO_OF_STACK_DUMPS
+    echo INTERVAL_BETWEEN_DUMPS=$INTERVAL_BETWEEN_DUMPS
+    echo GET_EVERYTHING=${GET_EVERYTHING}
+  fi
+}
+
+function collect_data {
+  host=$1
+  wd=$2
+  srv_num=$3
+
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Collecting data for process running on ${host} with working_dir ${wd}"
+  fi
+
+  # make a sub directory for this host and pid
+  out_dir="${PWD}/debug_data/${host}-${srv_num}"
+
+  collector_host="pnq-kneeraj3"
+  mkdir -p $out_dir 
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Args Being passed"
+    echo "arg1 wd = ${wd}" 
+    echo "arg2 num_stack_dumps = ${NO_OF_STACK_DUMPS}" 
+    echo "arg3 interval_dumps = ${INTERVAL_BETWEEN_DUMPS}" 
+    echo "arg4 GET_EVERYTHING = ${GET_EVERYTHING}" 
+    echo "arg5 collector_host = ${collector_host}" 
+    echo "arg6 out dir = ${out_dir}"
+  fi
+
+  typeset -f | ssh $host "$(cat);collect_on_remote ${wd} ${NO_OF_STACK_DUMPS} \\
+      ${INTERVAL_BETWEEN_DUMPS} ${GET_EVERYTHING} ${collector_host} ${out_dir} ${VERBOSE}"
+}
+
+function collect_on_remote {
+  data_dir=$1
+  num_stack_dumps=$2
+  int_stack_dumps=$3
+  get_all=$4
+  collector_host=$5
+  collector_dir=$6
+  verbose=$7
+
+  # first get the pid. The latest log file with the header will have the pid
+  host=`hostname`
+  if [ ! -d $data_dir ]; then
+    echo "${data_dir} not found on host: ${host}"
+  fi
+
+  cd $data_dir
+  files=()
+  last_restart_log=""
+  for l in $( ls -t *.log )
+  do
+    copyright_headers=`grep 'Copyright (C)' ${l}`
+    if [ ! -z "$copyright_headers" ]; then
+      # also check for the pid line and get the pid
+      proc_id=`sed -n 's/.*Process ID: \([0-9]\+\)$/\1/p' ${l}`
+      if [ "${verbose}" = "1" ]; then
+        echo "Addiing file ${l} to the array"
+      fi
+      files+=($l)
+      last_restart_log=$l
+      break
+    fi
+  done
+
+  if [ -z "${proc_id}" ]; then
+    echo "No valid process id could be obtained from logs on ${host}"
+    exit 1
+  fi
+
+  # get the stack dumps first
+  dump_num=1
+  for i in {1..${num_stack_dumps}}
+  do
+    dump_num=`expr ${dump_num} + 1`
+    kill -URG $proc_id
+    sleep $int_stack_dumps
+  done
+
+  if [ "${get_all}" = "1" ]; then
+    if [ "${verbose}" = "1" ]; then
+      rsync -avrz $data_dir "${collector_host}:${collector_dir}/"
+    else
+      rsync -avrzq $data_dir "${collector_host}:${collector_dir}/"
+    fi
```

Comment:
Won't it be better to tar gzip it and then ship? That can be provided to us directly instead of user having to tar it:

```sh
tar cvzf "${data_dir}.tar.gz" "$data_dir" && \
rsync -av "${data_dir}.tar.gz" "${collector_host}:${collector_dir}/" && \
rm -f "${data_dir}.tar.gz"
```